### PR TITLE
Add meta attribute to the pypsa network object

### LINKED ISSUE #291: Add a network attribute `n.meta` to store e.g. configs in dictionary
It would be useful to have an `n.meta` attribute that is initialised with an empty dictionary.

This could for example store information on whether the network was solved successfully or the `config.yaml` from PyPSA-Eur(-Sec).

Question to solve: How good can dictionaries be stored in NetCDF format.

### PR DESCRIPTION
**Pull request by FabianHofmann**
Closes #291

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [x] Unit tests for new features were added (if applicable).
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.
- [x] I consent to the release of this PR's code under the MIT license.

### CODE REVIEW

File: test/test_io.py [hunk 1]
```diff
@@ -8,20 +8,28 @@
 
 def test_netcdf_io(scipy_network, tmpdir):
```
Comment (fneum) [inline test/test_io.py:9]: maybe also test empty and layered dictionaries:

```suggestion
import pytest

@pytest.mark.parametrize("meta", [{"test": "test"}, {"test": {"test": "test"}}])
def test_netcdf_io(scipy_network, tmpdir, meta):
```

(analogous also for hdf5 and csv IO tests)

**Review by fneum — COMMENTED** [review]
Slick. Even layered dictionaries work :) Would suggest to add a test for that. Then, good to go in.
